### ISSUE TITLE
Update docs to use mamba instead of conda

### ISSUE THREAD

**Issue by maresb** [iterative/dvc.org#2370]
# Bug Report

## Description

The installation docs recommend using `conda` for installing DVC in a conda environment. Unfortunately the `conda` program is so slow that it is barely usable, and its solver chokes very easily. It takes a ridiculous amount of time to install, as demonstrated iterative/dvc#5625 and iterative/dvc#5787, to the point where users believe it is a bug.

The modern solution to this problem is [mamba](https://github.com/mamba-org/mamba), which is a relatively new reimplementation of `conda` in C++.

### Reproduce

Steps for reproducing and remediating the bug can be found in iterative/dvc#5625.

### Expected

I expect that the documentation recommends a reasonably efficient means of installing DVC within a conda environment, for example with `mamba`.

One potential way to resolve this issue would be to change the `conda` command for Mac OS, Windows, and Linux from

```bash
conda install -c conda-forge dvc
```
to
```bash
conda install -c conda-forge mamba
mamba install -c conda-forge dvc
```

**Comment by pmrowla** [comment]
Moved this issue to the docs repo, adding a note regarding `mamba` installation seems like the most reasonable solution here

**Comment by maresb** [comment]
I asked on Gitter, and the conda-forge folks [do seem to support](https://gitter.im/conda-forge/conda-forge.github.io?at=60742f23ae90f3684077029e) updating the conda-forge docs to feature `mamba` more prominently.

**Comment by jorgeorpinel** [comment]
@maresb would you be willing to suggest an edit? You can go to a page like https://dvc.org/doc/install/macos and click [Edit on GitHub] to get started. See also https://dvc.org/doc/user-guide/contributing/docs.

I'm not sure we want to change the main `conda` instructions though, maybe just add a link to a `mamba` usage page for now?

**Comment by maresb** [comment]
I can't get to it right away, but I'll see what I can do, maybe this weekend or so. If someone else can do it sooner, then please go ahead.

Since nobody has yet seemed to have managed to clock how long it actually takes to install DVC with `conda`, I left it running today in the background.

`conda`: 7 hours, 57 minutes, 50 seconds.
`mamba`: 33 seconds.
(Factor: 868.8×)

My question is then:
#### Given this benchmark, does it make any sense at all to recommend using `conda`? Or should we recommend exclusively `mamba`?


As for methodology, ...<details> I tested `conda` with
```bash
docker run --rm -it continuumio/miniconda3

date ; conda install -y -c conda-forge dvc ; date
```
and similarly `mamba` with
```bash
docker run --rm -it continuumio/miniconda3

conda install -y mamba
date ; mamba install -y -c conda-forge dvc ; date
```
Runtime was computed by subtracting the dates. My host machine is Ubuntu 20.04 on an Intel Skylake Core i7-6700 / 3.4 GHz. I was using my computer today, but never for anything computationally intensive.

</details>

I'd be curious to see how long it takes on proprietary OSes, but I cannot benchmark those.

**Comment by shcheklein** [comment]
Thanks @maresb for the research! I've updated the files for now the way you suggested:

```bash
$ conda install -c conda-forge mamba
$ mamba install -c conda-forge dvc
```

I guess the only thing that is left here is to may be mention (one sentence) why do we install `mamba` in the first place, or a comment:

```bash
$ conda install -c conda-forge mamba # mamba speedups install 800x
$ mamba install -c conda-forge dvc
```

**Comment by maresb** [comment]
@shcheklein thanks for making this happen!

Looking at the webpage layout, there is not much room to provide details. From editing the webpage, here's what looks good to me:
![image](https://user-images.githubusercontent.com/15216687/114606996-474d1f80-9c9c-11eb-9428-9e00149f15ca.png)


I noticed that there is an additional `conda install` command in `README.md`.

By the way, the recent builds of DVC on `conda-forge` support Python 3.9. I strongly suspect that the message underneath was meant to imply "Currently, Python 3.9 is unsupported", so that could be removed (or at least bumped 3.8→3.9)

**Comment by shcheklein** [comment]
@maresb happy to help. This is amazing improvement!

> I noticed that there is an additional conda install command in README.md.

Good catch. Done.

> so that could be removed 

also, good catch - removing.

Added also that comment `# installs much faster than conda` . It'll be automatically redeployed soon. 

Feel free to close the ticket after that if you feel that this is enough :)

thanks again for this improvement!

**Comment by maresb** [comment]
I'm very glad to see this resolved. Ironically, I have yet to use DVC because when I tried to install it a year or so ago, I ran into all these installation problems. But I hear great things about it, and I look forward to trying it now that it works with my `mamba` environment! :wink:

One further minor point is that the conda-forge shield seems to be down: https://img.shields.io/conda/dn/conda-forge/dvc
I looked briefly, and I suspect the issue is on the side of shields.io, so maybe it will resolve itself in some time.

I'll close out this issue now. Thanks so much for the efficient response!!!
